src/linearmixedmodel.jl

```diff
@@ -65,6 +65,20 @@ function LinearMixedModel(f::FormulaTerm, tbl::Tables.ColumnTable;
         end
     end
     push!(feterms, FeMat(y, [""]))
+
+    # detect and combine RE terms with the same grouping var
+    grps = Set([fname(rt) for rt in reterms])
+    if length(grps) != length(reterms)
+        # this reduction step could be expensive for large models
+        reterms_simplified = ReMat{T}[]
+        for gg in grps
+            combined = amalgamate([ret for ret in reterms if fname(ret) == gg])
+            push!(reterms_simplified, combined)
+        end
+        reterms = reterms_simplified
+    end
+
+
```

Comment:
Let's go ahead and merge the PR like this.  I may end up rewriting this and the _amalgamate function later using the [`BlockDiagonals`](https://github.com/invenia/BlockDiagonals.jl) package later, if that's okay with you.

Reply:
Sounds like a good idea.